packages/util/src/clone-element-with-ref.js

```diff
@@ -0,0 +1,17 @@
+import { cloneElement } from 'react'
+
+import { isFunction, isRef } from './index.js'
+
+export function cloneElementWithRef(element, outerRef, props) {
+  const { ref } = element
```

Comment:
Huh, I didn't know you could get the ref from react elements like this.  I'm surprised this is exposed.  :themoreyouknow: